features/mbedtls/platform/inc/platform_mbed.h

```diff
@@ -20,3 +20,11 @@
 #if defined(DEVICE_TRNG)
 #define MBEDTLS_ENTROPY_HARDWARE_ALT
 #endif
+
+/*
+ * Enable mbed TLS to use mbed OS mutexes
+ */
+#if defined(MBEDTLS_THREADING)
```

Comment:
I like the idea of separating the namespaces, but I think we should stay consistent while doing so: we should rename the hardware acceleration macro accordingly (now it is still in the library's namespace `MBEDTLS_HW_SUPPORT`)

I would suggest the `MBED_TLS_` prefix. This way we can take into account that mbed TLS won't always be the TLS stack in an mbed OS application, respect the application developers choice and avoid confusions. I would omit the `CONFIG` part, because feel it a little bit redundant, since the aim of all of these macros is to configure the mbed TLS library within mbed OS. Also I think that if the macro names are shorter, then they are less intimidating and easier to use.